data/projects/improve_installation.md

```diff
@@ -18,30 +18,72 @@ Each bear has a REQUIREMENTS tuple which holds instances of requirement classes,
 found here (https://gitlab.com/coala/package_manager), which should be
 reworked so that most operations in cib would be methods from these classes.
 
-As a start for this project, EACH bear must have all its dependencies solved
-out (>80% do), including the platform dependent ones. For this to be done,
-DistributionRequirement should be reworked.
+The problem right now is that some bears are not even getting installed by
+cib. Besides those with which we have requirement classes, the rest
+aren't even treated. So the plan is to create package managers for the
+popular managers, and treat them all in an universal DistributionRequirement.
+
+How this will work: Each requirement class will have an "install_command" and
+an "install" method. We will just call requirement.install() and this will
+run the command on its' own, not caring which platform you're on. Ideally,
+requirement.install() will use "install_command" to generate the command and
+run it with "sarge". For the DistributionRequirement class, the
+"install_command" method will check on which platform you are, and it will
+generate the according command.
+If the package is named differently on each platform, it shall be specified
+in the constructor. Take a mock example for a random bear:
+
+REQUIREMENTS=(DistributionRequirement(
+  apt-get='csslint', dnf='csslinter', brew='csslt')
+
+If you're on Ubuntu, it will run "apt-get install csslint". If you're on
+Fedora, it will run "dnf install csslinter", and so on.
+If your package manager is not on the list, we should just print a warning
+message so that the user tries to install it manually.
+
+So cib should not care what kind of requirement each is, it should just
+iterate through requirement instances and run "install_command()" on each,
+the requirement class handling the rest.
+
+Another special case is made of bears which hold different installations
+for each distribution. A good example is DartLintBear: it has different
+installation options for Linux (a wget script), Windows and Mac.
+https://www.dartlang.org/install
+
+A good idea to treat these is Conda: a conda package contains a script
+for each operating system (a .sh for Mac/Linux and a .bat for Windows).
+When you try to install the conda package, it will run the according script.
+
+The last special case is made out of bears which have bear dependencies.
+We have quite a few (the Clang bears). For these, we should have a
```

Comment:
Sentence is not complete, make it `We have quite a few of them (the Clang bears).`